impeller/renderer/allocator.cc

```
@@ -36,4 +40,22 @@ std::shared_ptr<DeviceBuffer> Allocator::CreateBufferWithCopy(
   return CreateBufferWithCopy(mapping.GetMapping(), mapping.GetSize());
 }
 
+std::shared_ptr<DeviceBuffer> Allocator::CreateBuffer(
+    const DeviceBufferDescriptor& desc) {
+  return OnCreateBuffer(desc);
+}
+
+std::shared_ptr<Texture> Allocator::CreateTexture(
+    const TextureDescriptor& desc) {
+  const auto max_size = GetMaxTextureSizeSupported();
+  if (desc.size.width > max_size.width || desc.size.height > max_size.height) {
+    VALIDATION_LOG
+        << "Requested texture size exceeds maximum supported size of "
+        << desc.size;
+    return nullptr;
```

Should we return an error enum?